test/models/ui/challenge_card_data_test.dart

```diff
@@ -0,0 +1,43 @@
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/models/ui/challenge_card_data.dart";
+
+void main() {
+  group("Challenge card data testing", () {
+    test("hash overrides correctly", () {
+      const challengeCard = ChallengeCardData(
+        title: "title",
+        distance: 50,
+        timeLeft: null,
+        reward: 100,
+      );
+
+      final expectedHash = Object.hash(
+        challengeCard.title,
+        challengeCard.distance,
+        challengeCard.timeLeft,
+        challengeCard.reward,
+      );
+
+      final actualHash = challengeCard.hashCode;
+
+      expect(actualHash, expectedHash);
+    });
+
+    test("equality overrides correctly", () {
+      const challengeCardData = ChallengeCardData(
+        title: "title",
+        distance: 50,
+        timeLeft: null,
+        reward: 100,
+      );
+      const challengeCardDataCopy = ChallengeCardData(
+        title: "title",
+        distance: 50,
+        timeLeft: null,
+        reward: 100,
+      );
+
+      expect(challengeCardData, challengeCardDataCopy);
```

Comment:
It seems that if we test equality with `expect` this part does not seems to be covered.

(This is also something that Yoan and I encountered in our PR. I think that dart is doing some optimizations and consider both objects as `identical`). The way I solved it was by testing inequalities on my object (73a13a8). I don't think it is important now, but it can make sure that our equalities is working well.

<img width="494" alt="image" src="https://github.com/ProximaEPFL/proxima/assets/53620532/a32d43e5-21fb-4926-8d07-051af1e32b26">

Reply:
Replacing const with var or final might also fix this create and two objects in memory.

Reply:
I just tried and it seems that our dart code analyser does not like it. It prefer to use of `const` for performances...

I think this may make the code not pass on the CI. Which is sad. It was a good idea !

Reply:
After some discussions with @CHOOSEIT, we simply decided to remove the `identical` function, since it will, in fact, almost always return false. This follows [Dart's documentation](https://dart.dev/tools/linter-rules/hash_and_equals).
Fixed in fffdb4d9159282864705a89c1bdcf8590d12e96f.